### DISCUSSION TITLE
KafkaMirror2

### DISCUSSION THREAD

**Discussion by AEncinas19** [strimzi/strimzi-kafka-operator#7843] [Q&A]
Hi, I'm trying to deploy a KafkaMirror2 in microk8s.
I've executed microk8s kubectl apply -f kafka-mirror-maker-2.yaml -n my-kafka-project (my-kafka-project is the namespace where I've created my source-cluster and target-cluster) and the mirror maker doesn't get into ready state. It's always on desired state

Also, I've applied all de CRDs like the quickstart indicates.

I need to install anything more? Is it a problem of the cluster operator?

Thank you in advance.

**Comment by scholzj** [comment]
TBH. without you sharing the configurations and logs, nobody can really answer this.

**Reply by AEncinas19** [reply]
Oh, sorry. Here are the configurations,
My Kafka clusters have the same configuration and are been deployed in the same namespace, but the second cluster's name is my-cluster2:
apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: my-cluster
spec:
  kafka:
    version: 3.3.1
    replicas: 4
    listeners:
      - name: plain
        port: 9092
        type: internal
        tls: false
      - name: external
        port: 9094
        type: nodeport
        tls: false
    storage:
      type: jbod
      volumes:
      - id: 0
        type: persistent-claim
        size: 100Gi
        deleteClaim: false
    config:
      offsets.topic.replication.factor: 3
      transaction.state.log.replication.factor: 3
      transaction.state.log.min.isr: 2
      default.replication.factor: 3
      min.insync.replicas: 2
      inter.broker.protocol.version: "3.3"
  zookeeper:
    replicas: 2
    storage:
      type: persistent-claim
      size: 100Gi
      deleteClaim: false
  entityOperator:
    topicOperator: {}
    userOperator: {}

Here is the KafkaMirror configuration:
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaMirrorMaker2
metadata:
  name: my-mirror-maker-2
spec:
  version: 3.3.1
  replicas: 1
  connectCluster: "my-cluster2"
  clusters:
    - alias: "my-cluster"
    bootstrapServers: my-cluster-kafka-bootstrap:9092
    - alias: "my-target-cluster"
    bootstrapServers: my-cluster2-kafka-bootstrap:9092
    config:
      # -1 means it will use the default replication factor configured in the broker
      config.storage.replication.factor: -1
      offset.storage.replication.factor: -1
      status.storage.replication.factor: -1
  mirrors:
    - sourceCluster: "my-source-cluster"
    targetCluster: "my-target-cluster"
    sourceConnector:
      config:
        replication.factor: 1
        offset-syncs.topic.replication.factor: 1
        sync.topic.acls.enabled: "false"
    heartbeatConnector:
      config:
        heartbeats.topic.replication.factor: 1
    checkpointConnector:
      config:
        checkpoints.topic.replication.factor: 1
    topicsPattern: ".*"
    groupsPattern: ".*"

The entity operator is the file 060-Deployment-strimzi-cluster-operator.yaml.

I did kubectl apply -f <Mirrorfilename> -n my-kafka-project and the console show that it was created, but when I execute kubectl get KafkaMirrorMaker2 the output shows this:
 
![Captura de pantalla de 2022-12-22 15-29-38](https://user-images.githubusercontent.com/49205996/209156307-fdfa0b58-e9fc-40ee-940f-785dd52e7e95.png)

**Reply by scholzj** [reply]
Can you plase format it as code so that the YAMLs can be read? Also, the logs from the cluster operator, Kafka brokers and the Mirror Maker pod will be needed as well.

One thing I can see already now is that this `sourceCluster: "my-source-cluster"` is wrong ... you used the alias `my-cluster` instead of `my-source-cluster`. So maybe you can try to first fix this if it helps before providing the other things.

**Reply by AEncinas19** [reply]
Solved. Thank you very much.
I feel stupid.

**Reply by scholzj** [reply]
Glad it is solved. And don't feel stupid ... we all have moments like this ;-).
